src/UI/templates/js/Modal/modal.js

```diff
@@ -25,15 +29,48 @@ il.UI = il.UI || {};
                 var $modal = $('#' + id);
                 $modal.modal(options);
             }
-        };
+			initializedModalboxes[signalData.id] = id;
+		};
 
         var closeModal = function (id) {
             $('#' + id).modal('close');
         };
 
+        /**
+         * Replace the content of the modalbox showed by the given showSignal with the data returned by the URL
+         * set in the signal options.
+         *
+         * @param showSignal ID of the show signal for the modalbox
+         * @param signalData Object containing all data from the replace signal
+         */
+        var replaceContentFromSignal = function (showSignal, signalData) {
+
+            // Find the ID of the triggerer where this modalbox belongs to
+            var triggererId = (showSignal in initializedModalboxes) ? initializedModalboxes[showSignal] : 0;
+            if (!triggererId) return;
+
+            // Find the content of the modalbox
+            var $modal = $('#' + triggererId + " .modal");
+            var url = signalData.options.url;
+
+            // get new stuff via ajax
+			$.ajax({
+				url: url,
+				dataType: 'html'
+			}).done(function(html) {
+				var $new_modal = $("<div>" + html + "<div>");
+
+                // of the new modal, we want the inner html of the modal (without the new top modal node, since
+                // we want to keep our id. Additionally we want the script tag with its content.
+                // Since html() gives us the inner html of the script tag only, we clone, wrap and get the inner from the wrapper...
+				$modal.html($new_modal.find(".modal").first().html() + $new_modal.find("script").first().clone().wrap('<p/>').parent().html());
```

Comment:
Mark by a css class? "ilReplaceNode", "ilReplaceContentNode", "ilReplaceFooterNode" with general signale ReplaceSignal, ReplaceContentSignal, ReplaceFooterSignal? Where to put the js?

Reply:
I would prefer that over the dependency on the structure. The js then could go somewhere near the new location for Signals? IMO this could also be done after this PR.